components/SwapPageContent/SwapPageContent.tsx

```diff
@@ -113,30 +113,35 @@ export function SwapPageContent() {
 
   const sqrtPriceAfter = useAsyncValue(async () => {
     if (!amounts) return null;
-    if (amounts.tradeType === TradeType.EXACT_INPUT) {
+    const paprIn = paprTokenField === Field.INPUT;
+    const paprOut = paprTokenField === Field.OUTPUT;
+
+    if (paprIn) {
       const quoteResult = await getQuoteForSwap(
         ethers.utils.parseUnits(
           amounts.input.toExact(),
           amounts.input.currency.decimals,
         ),
         amounts.input.currency.wrapped.address,
-        amounts.output.currency.wrapped.address,
+        underlying.id,
         tokenName as SupportedToken,
       );
       return quoteResult.sqrtPriceX96After;
-    } else {
+    } else if (paprOut) {
       const quoteResult = await getQuoteForSwapOutput(
         ethers.utils.parseUnits(
           amounts.output.toExact(),
           amounts.output.currency.decimals,
         ),
-        amounts.input.currency.wrapped.address,
+        underlying.id,
         amounts.output.currency.wrapped.address,
         tokenName as SupportedToken,
       );
       return quoteResult.sqrtPriceX96After;
+    } else {
+      return null;
```

Comment:
do we need to clear any state here? set something to null so the predicated rate doesn't show if, e.g. they select to swap USDC to ETH?

Reply:
we already do that in the handle token swap section